src/vs/workbench/contrib/terminal/browser/terminalWidgetManager.ts

```diff
@@ -91,13 +97,22 @@ class MessageWidget {
 	constructor(
 		private _container: HTMLElement,
 		private _left: number,
-		private _top: number,
-		private _text: string
+		private _y: number,
+		private _text: string,
+		private _verticalAlignment: WidgetVerticalAlignment
 	) {
 		this._domNode = document.createElement('div');
 		this._domNode.style.position = 'absolute';
 		this._domNode.style.left = `${_left}px`;
-		this._domNode.style.bottom = `${_container.offsetHeight - Math.max(_top, WIDGET_HEIGHT)}px`;
+
+		if (this.verticalAlignment === WidgetVerticalAlignment.Top) {
+			// Y position is to the top of the widget
+			this._domNode.style.bottom = `${Math.max(_y, WIDGET_HEIGHT) - WIDGET_HEIGHT}px`;
+		} else {
+			// Y position is to the bottom of the widget
+			this._domNode.style.bottom = `${Math.min(Math.max(_y, WIDGET_HEIGHT), _container.offsetHeight - WIDGET_HEIGHT)}px`;
```

Comment:
Looks like this is the problem, we need the row height to be passed through and used for the very bottom row.

Reply:
We don't actually want to check the max of `_y` and `WIDGET_HEIGHT` when y is to the bottom of the widget, since even y = 0 will still be in the viewport.